scripts/support/create-support-bundle.sh

```diff
@@ -0,0 +1,320 @@
+}
+
+succeeded() {
+  increase_indent
+  success "Succeeded!"
+  decrease_indent
+}
+
+failed() {
+  error "Failed!"
+}
+
+root_check() {
+  system_user_name=$(id -un)
+  if [ "${system_user_name}" != 'root' ]
+  then
+    # If not root, ensure the running user has access to
+    # sudo.
+    if sudo -l | grep "${system_user_name}" >/dev/null; then
+        return
+    else
+        failed
+        error_exit "$LINENO" "Script needs to be run as root or with sudo"
+    fi
+  fi
+}
+
+os_check() {
+  info "Checking that the operating system is supported..."
+  os_type=$(uname -s)
+  case "$os_type" in
+    Linux)
+      succeeded
+      ;;
+    *)
+      failed
+      error_exit "$LINENO" "The operating system $(fg_yellow "$os_type") is not supported by this script."
+      ;;
+  esac
+}
+
+os_arch_check() {
+  info "Checking for valid operating system architecture..."
+  arch=$(uname -m)
+  case "$arch" in
+    x86_64|amd64)
+      arch=amd64
+      ;;
+    aarch64|arm64|aarch64_be|armv8b|armv8l)
+      arch=arm64
+      succeeded
+      ;;
+    *)
+      failed
+      error_exit "$LINENO" "The operating system architecture $(fg_yellow "$arch") is not supported by this script."
+      ;;
+  esac
+}
+
+
+# This will check if the current environment has
+# all required shell dependencies to run the installation.
+dependencies_check() {
+  info "Checking for script dependencies..."
+  FAILED_PREREQS=''
+  for prerequisite in $PREREQS; do
+    if command -v "$prerequisite" >/dev/null; then
+      continue
+    else
+      if [ -z "$FAILED_PREREQS" ]; then
+        FAILED_PREREQS="${fg_red}$prerequisite${reset}"
+      else
+        FAILED_PREREQS="$FAILED_PREREQS, ${fg_red}$prerequisite${reset}"
+      fi
+    fi
+  done
+
+  if [ -n "$FAILED_PREREQS" ]; then
+    failed
+    error_exit "$LINENO" "The following dependencies are required by this script: [$FAILED_PREREQS]"
+  fi
+  succeeded
+}
+
+check_prereqs() {
+  banner "Checking Prerequisites"
+  increase_indent
+  root_check
+#   os_check
+  os_arch_check
+  dependencies_check
+  success "Prerequisite check complete!"
+  decrease_indent
+}
+
+function bundle_files() {
+    banner "Collecting files for support bundle"
+    increase_indent
+    # Directory for logs
+    log_dir="$collector_dir/log"
+    
+    # Check if directory exists
+    if [ ! -d "$log_dir" ]; then
+        info "Directory ($fg_red "$log_dir")$(reset) does not exist."
+        read -p "Please enter an existing directory for logs: " log_dir
+        if [ ! -d "$log_dir" ]; then
+            echo "Directory $log_dir does not exist."
+            return 1
+        fi
+    fi
+
+    read -p "Do you want to include only the most recent logs (y or n)? " response
+    increase_indent
+    tar_filename="support_bundle_$(date +%Y%m%d_%H%M%S).tar"
+    if [ "$response" = "n" ]; then
+        # Get all the log files
+        info "Collecting all log files in $(fg_cyan "$log_dir")$(reset)"
+        sudo tar -cvf $tar_filename $log_dir
```

Comment:
Can we hide the output of the tar command. I ran this an it's a bit jarring from the cleaner experience.

```sh
Do you want to include only the most recent logs (y or n)? n
    Collecting all log files in /opt/observiq-otel-collector/log
tar: Removing leading `/' from member names
/opt/observiq-otel-collector/log/
/opt/observiq-otel-collector/log/collector.log
tar: Removing leading `/' from member names
tar: Removing leading `/' from hard link targets
    Added file /etc/issue to the tar file.
tar: Removing leading `/' from member names
tar: Removing leading `/' from hard link targets
    Added file /etc/os-release to the tar file.
```